frontend/src/bundles/studio/components/timeline/timeaxis.tsx

```diff
@@ -0,0 +1,115 @@
+import { useTimelineContext } from 'dnd-timeline';
+
+import { Box } from '~/bundles/common/components/components.js';
+import { useMemo } from '~/bundles/common/hooks/hooks.js';
+
+type Marker = {
+    label?: string;
+    sideDelta: number;
+    heightMultiplier: number;
+};
+
+type MarkerDefinition = {
+    value: number;
+    maxRangeSize?: number;
+    minRangeSize?: number;
+    getLabel?: (time: Date) => string;
+};
+
+type TimeAxisProperties = {
+    markers: MarkerDefinition[];
+};
+
+const TimeAxis = (properties: TimeAxisProperties): JSX.Element => {
+    const { range, direction, sidebarWidth, valueToPixels } =
+        useTimelineContext();
+    const side = direction === 'rtl' ? 'right' : 'left';
+
+    const markers = useMemo(() => {
+        const sortedMarkers = properties.markers.toSorted(
+            (a, b) => b.value - a.value,
+        );
+        const delta = sortedMarkers.at(-1)?.value ?? 0;
+        const rangeSize = range.end - range.start;
+        const startTime = Math.floor(range.start / delta) * delta;
+        const endTime = range.end;
+        const timezoneOffset = new Date().getTimezoneOffset() * 60 * 1000;
```

Comment:
Avoid using magic numbers
B31 https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md

Reply:
Resolved as I returned to last commit (before deleting date-fns) as discussed in meeting.